include/oneapi/tbb/detail/_concurrent_queue_base.h

```diff
@@ -134,9 +134,9 @@ class micro_queue {
             } else {
                 head_page.store(p, std::memory_order_relaxed);
             }
-            tail_page.store(p, std::memory_order_release);
+            tail_page.store(p, std::memory_order_relaxed);
         } else {
-            p = tail_page.load(std::memory_order_acquire); // TODO may be relaxed ?
+            p = tail_page.load(std::memory_order_relaxed);
```

Comment:
Could you please explain why reading of `tail_page` with relaxed memory semantics is enough here?

Reply:
Got it! Thanks!